pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -354,6 +359,63 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 		}
 	}
 
+	// Inject CA bundle configMap if caBundleConfigMapName or constants.DefaultGlobalCaBundleConfigMapName annotation set
+	caBundleConfigMapName := mi.config.CaBundleConfigMapName
+	if ok := needCaBundelMount(caBundleConfigMapName, initContainer); ok {
+		if pod.Namespace != constants.KServeNamespace {
+			caBundleConfigMapName = constants.DefaultGlobalCaBundleConfigMapName
+		}
+
+		for _, envVar := range initContainer.Env {
+			if envVar.Name == s3.AWSCABundleConfigMap {
+				caBundleConfigMapName = envVar.Value
+				break
+			}
+		}
+
+		initContainer.Env = append(initContainer.Env, v1.EnvVar{
+			Name:  constants.GlobalCaBundleConfigMapNametEnvVarKey,
+			Value: caBundleConfigMapName,
+		})
+
+		caBundleVolumeMountPath := mi.config.CaBundleVolumeMountPath
+		if caBundleVolumeMountPath == "" {
+			caBundleVolumeMountPath = constants.DefaultCaBundleVolumeMountPath
+		}
+
+		for _, envVar := range initContainer.Env {
```

Comment:
there is 2 for loops, maybe check both envs on one for loop, no?

Reply:
updated